src/components/ImportForm/SourceSection/AuthOptions.tsx

```diff
@@ -30,7 +30,8 @@ const AuthOptions: React.FC = () => {
   const startAuthorization = React.useCallback(async () => {
     if (oAuthUrl) {
       const token = await getToken();
-      window.open(`${oAuthUrl}&k8s_token=${token}`, '_blank');
+      await initiateSpiAuthSession(oAuthUrl, token);
+      window.open(oAuthUrl, '_blank');
```

Comment:
Do you think it would make sense to make the init post call on load so that clicking auth button would be more instant?

Reply:
I tried this approach but if we click the button as soon as it is shown on UI there is a chance that the login request in the useEffect wont be completed and `oAuthUrl` in new tab will show error.